marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs

```diff
@@ -120,44 +120,48 @@ initialCoordinator indexerCount' =
 -- The points should/could provide shared access to the indexers themselves. The result
 -- is a list of points (rather than just one) since it offers more resume possibilities
 -- to the node (in the unlikely case there were some rollbacks during downtime).
-type Worker = SecurityParam -> Coordinator -> FilePath -> IO [Storable.StorablePoint ScriptTx.ScriptTxHandle]
+type Worker = SecurityParam -> Coordinator -> FilePath -> IO (Storable.StorablePoint ScriptTx.ScriptTxHandle)
 
 datumWorker :: Worker
 datumWorker securityParam Coordinator{_barrier, _channel, _errorVar} path = do
   ix <- Datum.open path (Datum.Depth $ fromIntegral securityParam)
   workerChannel <- atomically . dupTChan $ _channel
-  void . forkIO $ innerLoop workerChannel ix
-  pure [ChainPointAtGenesis]
-  where
-    innerLoop :: TChan (ChainSyncEvent (BlockInMode CardanoMode)) -> DatumIndex -> IO ()
-    innerLoop ch index = do
+  let
+    innerLoop :: DatumIndex -> IO ()
+    innerLoop index = do
       signalQSemN _barrier 1
       failWhenFull _errorVar
-      event <- atomically $ readTChan ch
+      event <- atomically $ readTChan workerChannel
       case event of
         RollForward blk _ct ->
-          Ix.insert (getDatums blk) index >>= innerLoop ch
+          Ix.insert (getDatums blk) index >>= innerLoop
         RollBackward cp _ct -> do
           events <- Ix.getEvents (index ^. Ix.storage)
-          innerLoop ch $
+          innerLoop $
             fromMaybe index $ do
               slot   <- chainPointToSlotNo cp
               offset <- findIndex (any (\(s, _) -> s < slot)) events
               Ix.rewind offset index
 
+    resumeFromCp = C.ChainPointAtGenesis
```

Comment:
I know you've just replicated the existing logic, but shouldn't we compute the resumePoint from the content of the indexer?

Reply:
The datum indexer seemed to be abandoned as it hasn't been updated to the newer Storable instance. Do you think I should write straight sql in here, or have custom helper in Datum's own module and use that?